service/tcp.go

```diff
@@ -149,7 +157,7 @@ func (s *tcpService) SetTargetIPValidator(targetIPValidator onet.TargetIPValidat
 	s.targetIPValidator = targetIPValidator
 }
 
-func dialTarget(tgtAddr socks.Addr, proxyMetrics *metrics.ProxyMetrics, targetIPValidator onet.TargetIPValidator) (onet.DuplexConn, *onet.ConnectionError) {
+func DefaultDialTarget(tgtAddr string, clientAddr net.Addr, proxyMetrics *metrics.ProxyMetrics, targetIPValidator onet.TargetIPValidator) (onet.DuplexConn, *onet.ConnectionError) {
```

Comment:
Don't you need the rest of the logic in this function?
How are you measuring the data in your custom dialer?

I need more context on your needs to figure out how to refactor the code here.

It seems like we may want a low level TCPDialer that is used for the first hop and adds measurement + perhaps keep alive. That TCP Dialer would be shared between a direct target dialer and your custom target dialer.